### ISSUE TITLE
Extend Azure DevOps and Gitlab as VCS

### ISSUE THREAD

**Issue by andylim0221** [aws-ia/terraform-aws-control_tower_account_factory#114]
It'd be great to extend VSC range for AFT to include Azure DevOps and Gitlab due to company tools or any other reasons and the developers are interested in extending this functionality in their VCS

**Comment by andylim0221** [comment]
Sorry, just noticed there is Gitlab support for this. Will change the title

**Comment by adam-daily** [comment]
Hey there, thanks for the suggestion. Currently we support all the VCS that CodeStar Connections supports (you can find them all here: https://docs.aws.amazon.com/dtconsole/latest/userguide/supported-versions-connections.html). I'll reach out to the CodeStar Connections product team though and let them know that there's interest to support Azure DevOps.

**Comment by jeremyciak** [comment]
+1

This would be great. I worked around this by adding custom Terraform to the setup/deployment for this module to configure Azure DevOps and AWS/CodeCommit so any changes in AzDO got replicated. Native support is always welcome and preferable.

**Comment by jkrusic19** [comment]
What about

**Comment by vaspoz** [comment]
It's not actually a limitation of AFT, but rather - CodePipeline. It allows only a limited number of VCS.
Here's what AWS suggests to do - [link](https://docs.aws.amazon.com/prescriptive-guidance/latest/patterns/use-third-party-git-source-repositories-in-aws-codepipeline.html#use-third-party-git-source-repositories-in-aws-codepipeline-tools)

> +1
> 
> This would be great. I worked around this by adding custom Terraform to the setup/deployment for this module to configure Azure DevOps and AWS/CodeCommit so any changes in AzDO got replicated. Native support is always welcome and preferable.

maybe it's the same as what you described, but here's the link with the "workaround" [implementation](https://aws.amazon.com/blogs/devops/use-aws-codecommit-to-mirror-an-azure-devops-repository-using-an-azure-devops-pipeline/).

**Comment by jeremyciak** [comment]
Yes, I used that workaround, but implemented everything in Terraform.

**Comment by vaspoz** [comment]
awesome! Could you share? I also need to implement that, but preferably via code

**Comment by hanafya** [comment]
AFT now supports GitLab and GitLab self-managed as version control systems (VCS) alternatives. [Latest Release](https://github.com/aws-ia/terraform-aws-control_tower_account_factory/releases/tag/1.13.2)

**Comment by richardsonky** [comment]
we also need azure devops support or we need ability to upload the artifacts to an s3 bucket instead to trigger the pipeline.
